HeartbeatLogging/Sources/StorageFactory.swift

```diff
@@ -16,37 +16,45 @@ import Foundation
 
 /// A factory type for `Storage`.
 protocol StorageFactory {
-  static func makeStorage(id: String) -> Self
+  static func makeStorage(id: String) -> Storage
 }
 
 // MARK: - FileStorage + StorageFactory
 
+/// The name of the file system directory where heartbeat data is stored.
+let kHeartbeatFileStorageDirectoryPath = "google-heartbeat-storage"
+
 extension FileStorage: StorageFactory {
-  static func makeStorage(id: String) -> FileStorage {
+  static func makeStorage(id: String) -> Storage {
     let rootDirectory = FileManager.default.applicationSupportDirectory
-    let storagePath = "google-heartbeat-storage/heartbeats-\(id)"
+    let heartbeatDirectoryPath = kHeartbeatFileStorageDirectoryPath
+    let heartbeatFilePath = "heartbeats-\(id)"
+
     let storageURL = rootDirectory
-      .appendingPathComponent(storagePath, isDirectory: false)
+      .appendingPathComponent(heartbeatDirectoryPath, isDirectory: true)
+      .appendingPathComponent(heartbeatFilePath, isDirectory: false)
 
     return FileStorage(url: storageURL)
   }
 }
 
-// MARK: - UserDefaultsStorage + StorageFactory
-
-extension UserDefaultsStorage: StorageFactory {
-  static func makeStorage(id: String) -> UserDefaultsStorage {
-    let suiteName = "com.google.heartbeat.storage"
-    let defaults = UserDefaults(suiteName: suiteName)
-    return UserDefaultsStorage(defaults: defaults, key: "heartbeats-\(id)")
+extension FileManager {
+  var applicationSupportDirectory: URL {
+    // TODO: The below bang! should be safe but re-evaluate.
+    urls(for: .applicationSupportDirectory, in: .userDomainMask).first!
   }
 }
 
-// MARK: - FileManager + Extension
+// MARK: - UserDefaultsStorage + StorageFactory
 
-private extension FileManager {
-  var applicationSupportDirectory: URL {
-    // TODO: The below bang! should be safe but re-evaluate.
-    urls(for: .applicationSupportDirectory, in: .userDomainMask).first!
+/// The name of the user defaults suite where heartbeat data is stored.
+let kHeartbeatUserDefaultsSuiteName = "com.google.heartbeat.storage"
```

Comment:
Nit: same message about the constant. It could be grouped with the above one

Reply:
Reworked in 29cf736